source/core/module_coreHelpers.js

```diff
@@ -303,37 +303,40 @@
             return currentFPS;
         };
 
-        var writeNewError = function (newErrorStatus, newErrorCode, newErrorSource, existingErrorStatusPointer, existingErrorCodePointer, exisitingErrorSourcePointer) {
-            Module.eggShell.dataWriteBoolean(existingErrorStatusPointer, newErrorStatus);
-            Module.eggShell.dataWriteInt32(existingErrorCodePointer, newErrorCode);
-            Module.eggShell.dataWriteString(exisitingErrorSourcePointer, newErrorSource);
+        var writeNewError = function (errorValueRef, newError) {
+            Module.eggShell.writeJSON(errorValueRef, JSON.stringify(newError));
         };
 
-        Module.coreHelpers.mergeErrors = function (newErrorStatus, newErrorCode, newErrorSource, existingErrorStatusPointer, existingErrorCodePointer, exisitingErrorSourcePointer) {
+        Module.coreHelpers.mergeErrors = function (errorValueRef, newError) {
             // Follows behavior of merge errors function: https://zone.ni.com/reference/en-XX/help/371361N-01/glang/merge_errors_function/
+            if (errorValueRef === undefined) {
+                return;
+            }
 
-            var existingErrorStatus = Module.eggShell.dataReadBoolean(existingErrorStatusPointer);
+            var errorStatusValueRef = Module.eggShell.findSubValueRef(errorValueRef, 'status');
+            var existingErrorStatus = Module.eggShell.readDouble(errorStatusValueRef) !== 0;
             var isExistingError = existingErrorStatus;
```

Comment:
You could declare variables as necessary always in this method. You seem to be doing in some places but not others. Move "var isNewError = newError.status;" below "if (isExistingError)"

Reply:
I had organized it so all the error checks are done in a group followed by all the warning checks (get info needed and then perform logic).

I thought mixing the acquiring of needed state and logic using the state was more confusing. Also var isNewError = newError.status is very little work, so it did not really add overhead.

It's just a stylistic choice to me, @sanmut if it's still clear and you prefer that style that's fine.

Reply:
Not a biggie. It is generally a good convention to declare variables as needed. If it helps with clarity,  then exceptions are ok.